Mlem/Views/Tabs/Search/Search View.swift

```diff
@@ -30,16 +30,39 @@ struct SearchView: View {
     
     @State private var navigationPath = NavigationPath()
     
+    @Namespace var scrollToTop
+    private var scrollToId: (any Hashable)? {
+        if showRecentSearches {
+            return scrollToTop
+        } else {
+            return communitySearchResultsTracker.foundCommunities.first?.id
+        }
+    }
+    
     // constants
     private let searchPageSize = 50
 
     var body: some View {
         NavigationStack(path: $navigationPath) {
-            content
-                .handleLemmyViews()
-                .navigationBarTitleDisplayMode(.inline)
-                .navigationBarColor()
-                .navigationTitle("Search")
+            ScrollViewReader { proxy in
+                content
+                    .handleLemmyViews()
+                    .navigationBarTitleDisplayMode(.inline)
+                    .navigationBarColor()
+                    .navigationTitle("Search")
+                    .onChange(of: selectedNavigationTabHashValue) { newValue in
+                        if newValue == TabSelection.search.hashValue {
+                            print("re-selected \(TabSelection.search) tab")
+                            if navigationPath.isEmpty, let scrollToId {
+                                withAnimation {
+                                    proxy.scrollTo(scrollToId, anchor: .bottom)
+                                }
```

Comment:
This should also switch focus to the search bar--that's a hotly requested feature (#246)